src/crypto/sign/ed25519.rs

```diff
@@ -66,6 +66,15 @@ pub fn gen_keypair() -> (PublicKey, SecretKey) {
     (pk, sk)
 }
 
+/// `sk_to_pk()` computes a public key from a secret key
+pub fn sk_to_pk(sk: &SecretKey) -> PublicKey {
```

Comment:
By the way, there is https://docs.rs/sodiumoxide/0.2.2/sodiumoxide/crypto/box_/curve25519xsalsa20poly1305/struct.SecretKey.html#method.public_key method in curve25519xsalsa20poly1305::SecretKey. Maybe you should do the same?

Reply:
No need to create a standalone function because there is a Universal Function Call Syntax in Rust: https://doc.rust-lang.org/1.7.0/book/ufcs.html